### PR TITLE
add SchedulerConfig to cyclegan training config

### PR DESCRIPTION
**Pull request by mcgibbon**
The CycleGAN model seems to perform better when it is able to train on a high learning rate to get into the neighbourhood of correct results, then decrease to lower learning rates where artifacting behaviours can be learned out of the model.

This PR adds a "scheduler" configuration option with SchedulerConfig to allow the user to configure an arbitrary learning rate scheduler to use in CycleGAN training.

Added public API:
- added SchedulerConfig to fv3fit.pytorch.optimizer

Refactored public API:
- added "scheduler" option to CycleGANTrainingConfig, which defaults to no learning rate scheduling

- [x] Tests added

Coverage reports (updated automatically):

### CODE REVIEW

File: external/fv3fit/fv3fit/pytorch/optimizer.py [hunk 1]
```diff
@@ -17,3 +20,27 @@ def instance(self, params):
...
+
+
+@dataclasses.dataclass
+class SchedulerConfig:
+    name: Optional[str] = None
+    kwargs: Mapping[str, Any] = dataclasses.field(default_factory=dict)
+
+    def instance(self, optimizer) -> Scheduler:
+        if self.name is None:
+            return NullScheduler()
```
Comment (elynnwu) [inline external/fv3fit/fv3fit/pytorch/optimizer.py:42]: [nit] this is technically not of type `Scheduler`, should it be `class NullScheduler(Scheduler)`?

**Review by elynnwu — APPROVED** [review]

File: external/fv3fit/fv3fit/pytorch/optimizer.py (hunk 1 above)
Comment (mcgibbon) [reply external/fv3fit/fv3fit/pytorch/optimizer.py:42]: Scheduler is a protocol type, meaning you don't have to subclass it to be an instance of it. `isinstance(null_scheduler, Scheduler)` would be True for `null_scheduler: NullScheduler`.
